src/Sarif.Multitool/Rules/SARIF2005.ProvideHelpfulToolInformation.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Json.Pointer;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool.Rules
+{
+    public class ProvideHelpfulToolInformation : SarifValidationSkimmerBase
+    {
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString
+        {
+            Text = RuleResources.SARIF2005_ProvideHelpfulToolInformation_FullDescription_Text
+        };
+
+        public override FailureLevel DefaultLevel => FailureLevel.Warning;
+
+        public override string Id => RuleId.ProvideHelpfulToolInformation;
+
+        protected override IEnumerable<string> MessageResourceNames => new string[]
+        {
+            nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_ProvideToolVersion_Text),
+            nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_ProvideConciseToolName_Text),
+            nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_UseNumericToolVersions_Text)
+        };
+
+        protected override void Analyze(Tool tool, string toolPointer)
+        {
+            if (tool == null)
+            {
+                return;
+            }
+
+            if (tool.Driver != null)
+            {
+                AnalyzeToolDriver(tool.Driver, toolPointer);
+            }
+        }
+
+        private void AnalyzeToolDriver(ToolComponent toolComponent, string toolPointer)
+        {
+            // ProvideConciseToolName: Ensure that tool.driver.name isn't more than 3 words long
+            if (!string.IsNullOrEmpty(toolComponent.Name))
+            {
+                const int maxWordCount = 3;
+                int wordCount = toolComponent.Name.Split(' ').Length;
+                if (wordCount > maxWordCount)
+                {
+                    string driverPointer = toolPointer.AtProperty("driver").AtProperty("name");
+                    LogResult(
+                        driverPointer,
+                        nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_ProvideConciseToolName_Text),
+                        wordCount.ToString(),
+                        maxWordCount.ToString());
+                }
+            }
+
+            // ProvideToolVersion: Either tool.driver.version or tool.driver.semanticVersion should be there.
+            if (string.IsNullOrWhiteSpace(toolComponent.Version) && string.IsNullOrWhiteSpace(toolComponent.SemanticVersion))
+            {
+                LogResult(
+                        toolPointer,
```

Comment:
>toolPointer [](start = 24, length = 11)

`toolDriverPointer`. The idea is that the bad object is the driver itself. You want to say "There's something wrong with tool/driver. What's wrong? It doesn't have a version."

Reply:
should i point to version? 

---
In reply to: [444574759](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444574759) [](ancestors = 444574759)

Reply:
Not on this one. The reason is that this `driver` object doesn't _have_ a `version` property. So the pointer would point nowhere. And again, the object that is _faulty_ is `tool.driver`. It's faulty because it doesn't have a version! So the message should point to the faulty object, `tool.driver`.

---
In reply to: [444576865](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444576865) [](ancestors = 444576865,444574759)